### PR TITLE
Accept regexps and arrays in allow_origin option

### PR DESCRIPTION
**Pull request by frm**
I'm not sure you want to include this but I just had to add support for regexps and lists in a project using this gem.

The use case was due to pull requests being deployed each in a different Netlify URL and those being different from production. My solution was to implement this and set `allow_origin` to `[/deploy-preview.*--myapp.netlify.app/, "https://production.my.app"]`.

### CODE REVIEW

**Comment by jdesrosiers** [comment]
This sounds like a great addition. Thanks! I'll review as soon as I can, but it probably won't be til the weekend.

File: spec/cors_spec.rb [hunk 1]
```diff
@@ -151,6 +151,16 @@ def make_request(allow_origin)
       make_request("*")
       expect(last_response["Access-Control-Allow-Origin"]).to eq("http://example.com")
     end
+
+    it "should allow regexps on the :allowed_origin option" do
```
Comment (jdesrosiers) [inline spec/cors_spec.rb]: The config should be `:allow_origin` not `:allowed_origin`. Same on line160.

**Review by jdesrosiers — CHANGES_REQUESTED** [review]
Sorry for the delay. This looks good. There's just a small typo in a test description. Also, would you mind updating the README to reflect the changes?

**Comment by frm** [comment]
Sorry it took me so long, just updated 👍

**Comment by jdesrosiers** [comment]
Thanks! New version has been deployed as v1.2.0

**Pull request merged**